apis/authentication/v1alpha1/openidconnect_types.go

```diff
@@ -116,6 +116,12 @@ type OIDCAuthenticationSpec struct {
 
 	// +optional
 
+	// ExtraClaims, if specified, causes the OIDCAuthenticator to copy list claims to the
+	// user Extra field.
+	ExtraClaims *string `json:"extraClaims,omitempty"`
```

Comment:
> As per [k8s](https://github.com/kubernetes/kubernetes/blob/29ff98ff372ed17405cf2c7f18d20653301a4218/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L39C2-L40) documentation the keys must be lowercase. Lets do this and also add validation in
> 
Thanks for pointing that out however, I haven't found anything similar for the claim name itself. Based on the [JWT RFC](https://www.rfc-editor.org/rfc/rfc7519.html#section-10.1.1), the claim name is case sensitive and should be short (under 8 char) if possible. Although most of the claims I have seen are following the lowercase with underscore for word separation format, there's no guarantee that people will do. However, using letter casing to distinguish two claims is strongly discourage thus what I would suggest is to simply lowercase the claims to generate the extra field key.

Reply:
I am OK with letting users set case sensitive claims and making them lowercase in the `extra` field. This will however create ambiguity if user sets 

```yaml
extraClaims:
- foo
- FoO
```

In order to avoid that we can introduce validation of this field which will prevent setting multiple claims that will be considered equal if compared case insensitively. The [validate](https://github.com/gardener/oidc-webhook-authenticator/blob/8b2f202660f63f6e8ccfdda2a5f19835cc943c96/apis/authentication/v1alpha1/openidconnect_webhook.go#L85) function can be extended to prevent such misconfigurations.

Reply:
Done as requested with a new test to ensure it's working as expected.